misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/PartialCoverageModifiedCounter.java

```
@@ -0,0 +1,82 @@
+package org.ballerinalang.test.runtime.entity;
+
+import org.jacoco.core.analysis.ICounter;
+
+/**
+ * Represents a counter that eliminates partially covered counter status and coverts it to fully covered status.
+ *
+ */
+public class PartialCoverageModifiedCounter implements ICounter {
+
+    private int covered;
+    private int missed;
+
+    public PartialCoverageModifiedCounter(ICounter prevCounter) {
+        this.covered = prevCounter.getCoveredCount();
+        this.missed = prevCounter.getMissedCount();
+        modifyCoverageNumbers();
+    }
+
+    /**
+     * Modify the covered and missed numbers in cases the counter status is calculated as PARTLY_COVERED.
+     * It converts the counter status to FULLY_COVERED.
+     */
+    public void modifyCoverageNumbers() {
+        if (getStatus() == PARTLY_COVERED) {
+            covered = covered + missed;
+            missed = 0;
+        }
+    }
+
+    @Override
+    public double getValue(CounterValue value) {
+        switch (value) {
+            case TOTALCOUNT:
+                return getTotalCount();
+            case MISSEDCOUNT:
+                return getMissedCount();
+            case COVEREDCOUNT:
+                return getCoveredCount();
+            case MISSEDRATIO:
+                return getMissedRatio();
+            case COVEREDRATIO:
+                return getCoveredRatio();
+            default:
+                throw new AssertionError(value);
```

Let's throw a runtime exception

---

Shall I throw an ```IllegalArgumentException```?